atlasdb-buggify/src/main/java/com/palantir/atlasdb/buggify/impl/DefaultBuggifyFactory.java

```diff
@@ -19,13 +19,33 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.palantir.atlasdb.buggify.api.Buggify;
 import com.palantir.atlasdb.buggify.api.BuggifyFactory;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
 import java.security.SecureRandom;
 import java.util.function.DoubleSupplier;
 
 public final class DefaultBuggifyFactory implements BuggifyFactory {
-    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
+    private static final SafeLogger log = SafeLoggerFactory.get(DefaultBuggifyFactory.class);
+    private static final SecureRandom SECURE_RANDOM;
     public static final BuggifyFactory INSTANCE = new DefaultBuggifyFactory();
 
+    static {
+        SecureRandom randomInstance;
+        try {
+            // Antithesis's implementation of /dev/urandom is deterministic and something controlled by their fuzzer,
+            // which allows them to explore the state space. We thus choose the NativePRNG algorithm which is backed by
+            // /dev/urandom. This string is a reserved identifier for the algorithm, following
+            // https://docs.oracle.com/en/java/javase/17/docs/specs/security/standard-names.html#securerandom-number-generation-algorithms
+            randomInstance = SecureRandom.getInstance("NativePRNG");
```

Comment:
Should we just make this a static utility, so we can also use it in the ring workflow and other places we will use random?

Reply:
Yeah we can do that. This is fine for the workflows, though note that in general injecting this in other places random is used might be hard (unless we want to put this in atlasdb-commons or something, which I wouldn't commit to unless we know we want to!)